bluemira/geometry/varied_offset.py

```diff
@@ -0,0 +1,165 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2022 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+Module containing functions to generate variable offset curves
+"""
+
+import numpy as np
+
+from bluemira.geometry.tools import find_clockwise_angle_2d, make_bspline
+from bluemira.geometry.wire import BluemiraWire
+
+
+def varied_offset(
+    wire: BluemiraWire,
+    min_offset: float,
+    max_offset: float,
+    min_offset_angle: float,
+    max_offset_angle: float,
+    num_points: int = 200,
+) -> BluemiraWire:
```

Comment:
Debating about whether it would be nice to perform the angle transitions (optionally) relative to the center-point of the widest x-z slice, rather than the COG. This is ill-defined though, so this isn't an actionable comment - just a musing!